### PR TITLE
Add support for .json files in node_modules

### PR DESCRIPTION
**Pull request by goshakkk**

### CODE REVIEW

File: lib/fs_utils/source_file.js [hunk 1]
```diff
@@ -114,7 +115,8 @@ const makeCompiler = (path, cache, linters, compilers, wrap) => {
 class SourceFile {
   constructor(path, compilers, linters, wrapper, isHelper, isVendor, fileList) {
     this.fileList = fileList;
-    const type = compilers && compilers[0] && compilers[0].type;
+    // treat json files from node_modules as javascript
+    const type = compilers && compilers[0] && compilers[0].type || deppack.isNpm(path) && 'javascript';
```
Comment (goshakkk) [inline lib/fs_utils/source_file.js]: not the best piece but then, not sure if there is a better way to let deppack influence file type...

File: lib/fs_utils/file_list.js [hunk 2]
```diff
@@ -195,7 +195,8 @@ class FileList {
       debug(`Reading ${path}`);
       fcache.updateCache(path, error => {
         if (error) throw new Error(formatError('Reading', error));
-        if (!ignored && (compiler && compiler.length)) {
+        // .json files from node_modules should always be compiled
+        if (!ignored && (compiler && compiler.length) || deppack.isNpm(path)) {
```
Comment (goshakkk) [inline lib/fs_utils/file_list.js]: same as below... json files from node modules won't have a compiler

Comment (paulmillr) [inline lib/fs_utils/file_list.js]: Can we ensure the code path is executed only for json files?

File: lib/fs_utils/file_list.js [hunk 3]
```diff
@@ -195,7 +195,8 @@ class FileList {
       debug(`Reading ${path}`);
       fcache.updateCache(path, error => {
         if (error) throw new Error(formatError('Reading', error));
-        if (!ignored && (compiler && compiler.length)) {
+        // .json files from node_modules should always be compiled
+        if (!ignored && (compiler && compiler.length) || deppack.isNpmJSON(path)) {
```
Comment (goshakkk) [inline lib/fs_utils/file_list.js]: > Can we ensure the code path is executed only for json files?

by being more explicit about function name, like this?

**Pull request merged**